fatal/type/sequence.h

```diff
@@ -1004,32 +1021,30 @@ struct polynomial<T, constant_sequence<T>, Variable, VariableAccumulator, Accumu
 
 template <typename T, std::size_t Size>
 constexpr std::size_t size(T const (&)[Size]) {
-  static_assert(
-    Size > 0,
-    "expecting a string containing at least the null terminator"
-  );
```

Comment:
```sh
eniebler-mbp:~ eniebler$ ~/llvm-install/bin/clang++ -pedantic -Wall -Wextra -std=gnu++11 -c test.cpp
test.cpp:2:14: warning: zero size arrays are an extension [-Wzero-length-array]
  int rg[] = {};
             ^
1 warning generated.
```

It's not legal C++. Looks like compilers implement this as an extension. I agree the assert should stay.